src/main/java/ch/njol/skript/conditions/CondIsLoaded.java

```diff
@@ -19,38 +19,93 @@
  */
 package ch.njol.skript.conditions;
 
-import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Direction;
+import ch.njol.util.Kleenean;
 
 import org.bukkit.Bukkit;
-import org.bukkit.Chunk;
+import org.bukkit.Location;
 import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
 
 @Name("Is Loaded")
-@Description("Checks whether or not a chunk/world is loaded")
-@Examples("if chunk at {home::%player's uuid%} is loaded:")
-@Since("2.3")
-public class CondIsLoaded extends PropertyCondition<Object> {
+@Description("Checks whether or not a chunk/world is loaded. 'chunk at 1, 1' uses chunk coords, which are location coords divided by 16.")
+@Examples({"if chunk at {home::%player's uuid%} is loaded:",
+		"if chunk 1, 10 in world \"world\" is loaded:",
+		"if world(\"lobby\") is loaded:"})
+@Since("2.3, INSERT VERSION (revamp with chunk at location/coords)")
+public class CondIsLoaded extends Condition {
 	
 	static {
-		register(CondIsLoaded.class, "loaded", "worlds/chunks");
+		Skript.registerCondition(CondIsLoaded.class,
+			"chunk[s] %directions% [%locations%] (is|are)[(1¦(n't| not))] loaded",
+			"chunk [at] %number%, %number% (in|of) [world] %world% is[(1¦(n't| not))] loaded",
+			"[world[s]] %worlds% (is|are)[(1¦(n't| not))] loaded");
 	}
 	
+	@Nullable
+	private Expression<Location> locations;
+	@Nullable
+	private Expression<Number> x,z;
+	@Nullable
+	private Expression<World> world;
+	@Nullable
+	private Expression<World> worlds;
+	private int pattern;
+	
+	@Override
+	public boolean init(Expression<?>[] exprs, int pattern, Kleenean isDelayed, ParseResult parseResult) {
+		locations = pattern == 0 ? Direction.combine((Expression<? extends Direction>) exprs[0], (Expression<? extends Location>) exprs[1]) : null;
+		x = pattern == 1 ? (Expression<Number>) exprs[0] : null;
+		z = pattern == 1 ? (Expression<Number>) exprs[1] : null;
+		world = pattern == 1 ? (Expression<World>) exprs[2] : null;
+		worlds = pattern == 2 ? (Expression<World>) exprs[0] : null;
```

Comment:
I believe the second world variable isn't needed

Reply:
What was I thinking? haha. My brain must have been elsewhere. Will be removed in next commit.